tests/backends/storage/test_s3.py

```diff
@@ -0,0 +1,349 @@
+
+    try:
+        s3()
+    except Exception:  # pylint:disable=broad-except
+        pytest.fail("S3Storage should not raise exception on successful connection")
+
+
+@mock_s3
+def test_backends_storage_s3_list_should_yield_archive_names(
+    moto_fs_compatibility, s3, fs
+):  # pylint:disable=unused-argument, invalid-name
+    """Tests that given S3Service.list method successfully connects to the S3
+    storage, the S3Storage list method should yield the archives.
+    """
+
+    # Regions outside of us-east-1 require the appropriate LocationConstraint
+    s3_client = boto3.client("s3", region_name="us-east-1")
+    bucket_name = "bucket_name"
+    s3_client.create_bucket(Bucket=bucket_name)
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key="2022-04-29.gz",
+        Body=json.dumps({"id": "1", "foo": "bar"}),
+    )
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key="2022-04-30.gz",
+        Body=json.dumps({"id": "2", "some": "data"}),
+    )
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key="2022-10-01.gz",
+        Body=json.dumps({"id": "3", "other": "info"}),
+    )
+
+    listing = [
+        {"name": "2022-04-29.gz"},
+        {"name": "2022-04-30.gz"},
+        {"name": "2022-10-01.gz"},
+    ]
+
+    history = [
+        {"id": "2022-04-29.gz", "backend": "s3", "command": "fetch"},
+        {"id": "2022-04-30.gz", "backend": "s3", "command": "fetch"},
+    ]
+
+    s3 = s3()
+    try:
+        response_list = s3.list()
+        response_list_new = s3.list(new=True)
+        response_list_details = s3.list(details=True)
+    except Exception:  # pylint:disable=broad-except
+        pytest.fail("S3Storage should not raise exception on successful list")
+
+    fs.create_file(settings.HISTORY_FILE, contents=json.dumps(history))
+
+    assert list(response_list) == [x["name"] for x in listing]
+    assert list(response_list_new) == ["2022-10-01.gz"]
+    assert [x["Key"] for x in response_list_details] == [x["name"] for x in listing]
+
+
+@mock_s3
+def test_backends_storage_s3_list_with_failed_connection_should_log_the_error(
+    moto_fs_compatibility, s3, fs, caplog
+):  # pylint:disable=unused-argument, invalid-name
+    """Tests that given S3Service.list method fails to retrieve the list of archives,
+    the S3Storage list method should log the error and raise a BackendException.
+    """
+
+    # Regions outside of us-east-1 require the appropriate LocationConstraint
+    s3_client = boto3.client("s3", region_name="us-east-1")
+    bucket_name = "bucket_name"
+    s3_client.create_bucket(Bucket=bucket_name)
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key="2022-04-29.gz",
+        Body=json.dumps({"id": "1", "foo": "bar"}),
+    )
+
+    s3 = s3()
+    s3.bucket_name = "wrong_name"
+
+    fs.create_file(settings.HISTORY_FILE, contents=json.dumps([]))
+    caplog.set_level(logging.ERROR)
+    error = "The specified bucket does not exist"
```

Comment:
```suggestion
    error = "The specified bucket does not exist."
```

Reply:
The error message for a ClientError from "list_objects_v2" seems to not have a dot at the end. It is indeed inconsistent with error message from "get_object" though!